pkg/oc/cli/cmd/set/deployment_strategy.go

```diff
@@ -0,0 +1,483 @@
+	cmd.Flags().Int64Var(&options.UpdatePeriodSeconds, "update-period-seconds", deployapi.DefaultRollingUpdatePeriodSeconds, "The time to wait between individual pod updates in rolling strategy.")
+	cmd.Flags().Int32Var(&options.Partition, "partition", 0, "A number that indicates the ordinal at which the StatefulSet should be partitioned.")
+	cmd.Flags().StringVar(&options.MaxUnavailable, "max-unavailable", "", "The maximum number of pods that can be unavailable during the update in rolling strategy.")
+	cmd.Flags().StringVar(&options.MaxSurge, "max-surge", "", "The maximum number of pods that can be scheduled above the original number of pods in rolling strategy.")
+	cmd.Flags().StringVar(&options.Image, "image", "", "An image which can carry out a custom deployment strategy.")
+	cmd.Flags().StringArrayVarP(&options.Environment, "environment", "e", options.Environment, "Environment variables to use in the custom deployment pod")
+
+	cmd.Flags().BoolVar(&options.Recreate, "recreate", options.Recreate, "Set a Recreate strategy")
+	cmd.Flags().BoolVar(&options.Rolling, "rolling", options.Rolling, "Set a Rolling strategy")
+	cmd.Flags().BoolVar(&options.Custom, "custom", options.Custom, "Set a Custom strategy")
+	cmd.Flags().BoolVar(&options.OnDelete, "on-delete", options.OnDelete, "Set a OnDelete strategy")
+
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set deployment hook will NOT contact api-server but run locally.")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter deployment configs")
+
+	// cmd.Flags().String("failure-policy", "ignore", "The failure policy for the deployment hook. Valid values are: abort,retry,ignore")
+
+	cmd.MarkFlagFilename("filename", "yaml", "yml", "json")
+	kcmdutil.AddDryRunFlag(cmd)
+
+	return cmd
+}
+
+func (o *DeploymentStrategyOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string) error {
+	resources := args
+	if i := cmd.ArgsLenAtDash(); i != -1 {
+		resources = args[:i]
+		o.Command = args[i:]
+	}
+	if len(o.Filenames) == 0 && len(args) < 1 {
+		return kcmdutil.UsageErrorf(cmd, "one or more resources must be specified as <name> or dc/<name>")
+	}
+
+	cmdNamespace, explicit, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.Cmd = cmd
+
+	mapper, _ := f.Object()
+	o.Builder = f.NewBuilder(!o.Local).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(explicit, &resource.FilenameOptions{Recursive: false, Filenames: o.Filenames}).
+		Flatten()
+	if !o.Local {
+		o.Builder = o.Builder.
+			ResourceNames("deploymentconfigs", resources...).
```

Comment:
needs other resources

Reply:
i believe this is just "default", iow. if you don't specify resource/ it defaults to DC.

Reply:
I am not sure why this is in `if !o.Local {` branch w.r.t. `cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set deployment hook will NOT contact api-server but run locally.")`

I don't think this command should default to any resource. Accept only the full syntax.

Reply:
> I don't think this command should default to any resource. Accept only the full syntax.

I agree, since this command does not deal with just a single resource kind, I don't think it would be good to implicitly assume a dc was specified from a given resource name

Reply:
We have same pattern established for `oc rollout` ;-) But I'm fine removing this defaulting.